view/rooms/app.js

```diff
@@ -935,8 +936,18 @@ window.onload = () => {
     const subTitle = `${_roomName} - ${res.body.name}`;
     uPage.setSubtitle(subTitle);
     document.title = subTitle;
-
-    if (room.settings && room.settings.botsLayout) {
+    let layoutCookie =
+      u.getCookie(`${window.location.pathname}-bots-layout`);
+    if (layoutCookie) {
+      layoutCookie = JSON.parse(layoutCookie);
+      // Checking if the layout that came from the cookie is valid
+      if (uPage.botLayoutIsValid(layoutCookie, room.bots)) {
+        _botsLayout = layoutCookie;
+        // Reordering bots so they will be in the correct layout
+        room.bots = _botsLayout.leftColumn.concat(_botsLayout.middleColumn)
+          .concat(_botsLayout.rightColumn);
+      }
```

Comment:
what happens if the boot is not valid?

Reply:
It will just use the default layout and wont use the one defined in the cookie